Extensions are failing to compile due to mismatch of the vscode-languageserver-types version in CircleCi

### Summary

We have seen a few build failures recently where the extensions are failing to compile.  
See https://app.circleci.com/pipelines/github/forcedotcom/salesforcedx-vscode/9000/workflows/1ece052c-3a2b-41c4-b5de-332f42f080ff/jobs/34461 for an example build failure. 

The build fails to compile with the following message: 
```
packages/salesforcedx-visualforce-language-server/src/visualforceServer.ts(310,39): error TS2345: Argument of type 'import("C:/Users/circleci/project/node_modules/vscode-languageserver-protocol/node_modules/vscode-languageserver-types/lib/umd/main").CompletionItem' is not assignable to parameter of type 'import("C:/Users/circleci/project/node_modules/vscode-languageserver-types/lib/main").CompletionItem'.
  Types of property 'kind' are incompatible.
    Type 'import("C:/Users/circleci/project/node_modules/vscode-languageserver-protocol/node_modules/vscode-languageserver-types/lib/umd/main").CompletionItemKind' is not assignable to type 'import("C:/Users/circleci/project/node_modules/vscode-languageserver-types/lib/main").CompletionItemKind'.
      Type '19' is not assignable to type 'CompletionItemKind'.
```

This type of failure usually occurs if the there are two different versions of a dependancy.  I ssh'd into the box for the build and found the following: 
```
PS C:\Users\circleci\project> cat /Users/circleci/project/node_modules/vscode-languageserver-protocol/node_modules/vscode-languageserver-types/package.json
cat /Users/circleci/project/node_modules/vscode-languageserver-protocol/node_modules/vscode-languageserver-types/package.json
{
	"name": "vscode-languageserver-types",
	"description": "Types used by the Language server for node",
	"version": "3.14.0",
	...
}
```
```
PS C:\Users\circleci\project> cat /Users/circleci/project/node_modules/vscode-languageserver-types/package.json
cat /Users/circleci/project/node_modules/vscode-languageserver-types/package.json
{
	"name": "vscode-languageserver-types",
	"description": "Types used by the Language server for node",
	"version": "3.4.0",
	...
}

So it appears that we have a large version gap when using the vscode-languageserver-types dependency in `packages/salesforcedx-visualforce-language-server/src/visualforceServer.ts`

I was unable to reproduce this locally and can see that the two dependancies both use v3.14.0 of the types module when doing a local build.  However the production build installs differently where it has all packages rolled up into the single node_modules folder.  I can not explain what change resulted in this error starting to occur. 

### Steps To Reproduce:

1.  Rerun a failed build in circleCI with ssh.
2. ssh into the circleci system.
3. Run `npm run compile`

### Expected result

Should compile successfully.

### Actual result

Error about `CompletionItemKind` and fails to compile.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -514,6 +514,10 @@ jobs:
       - checkout
       - restore_dependency_cache
       - gh-config
+      - run:
+          name: 'Install latest npm'
+          shell: bash.exe
+          command: nvm install-latest-npm
       - install
       - run:
           name: Install Lerna

